packages/devtools_app/lib/src/screens/network/network_request_inspector_views.dart

```diff
@@ -201,10 +207,122 @@ class HttpViewTrailingCopyButton extends StatelessWidget {
   }
 }
 
+/// A DropDownButton for selecting [NetworkResponseType].
+///
+/// If there is no content to visualise, the drop down will not show. Drop down
+/// values will update as the request's data is updated.
+class HttpResponseTrailingDropDown extends StatefulWidget {
+  const HttpResponseTrailingDropDown(
+    this.data, {
+    super.key,
+    required this.currentresponseType,
+    required this.onNewresponseTypeSelected,
+  });
+  final ValueListenable<NetworkResponseType> currentresponseType;
+  final DartIOHttpRequestData data;
+  final ValueChanged<String> onNewresponseTypeSelected;
+
+  @override
+  State<HttpResponseTrailingDropDown> createState() =>
+      _HttpResponseTrailingDropDownState();
+}
+
+class _HttpResponseTrailingDropDownState
+    extends State<HttpResponseTrailingDropDown> {
+  List<NetworkResponseType> availableResponseTypes = List.empty(growable: true);
+  bool visible = true;
+
+  @override
+  void initState() {
+    super.initState();
+    initialise();
+  }
+
+  @override
+  void didUpdateWidget(covariant HttpResponseTrailingDropDown oldWidget) {
+    initialise();
+    super.didUpdateWidget(oldWidget);
+  }
+
+  void initialise() {
+    availableResponseTypes.clear();
+    visible = true;
+    if ((widget.data.contentType != null &&
+            widget.data.contentType!.contains('image')) ||
+        widget.data.responseBody!.isEmpty) {
+      visible = false;
+    }
+    if (visible) {
+      availableResponseTypes.add(NetworkResponseType.auto);
+      if (isJsonDecodable()) {
+        availableResponseTypes.add(NetworkResponseType.json);
+      }
+      availableResponseTypes.add(NetworkResponseType.text);
```

Comment:
this can also be written as 
```dart
availableResponseTypes = [
  NetworkResponseType.auto,
  if (isJsonDecodable()) NetworkResponseType.json,
  NetworkResponseType.text,
];
```